### ISSUE TITLE
Server Error FirebaseError: Failed to get document because the client is offline. 

### ISSUE THREAD

**Issue by mikias21** [firebase/firebase-js-sdk#4722]
I am using firebase 8.3.2 with next js, when ever I do ssr I keep getting  Server Error Firebase Error: Failed to get document because the client is offline. error  can you please tell me what the problem is ?

**Comment by thebrianchen** [comment]
@mikias21  Thanks for writing in! The error you see usually occurs when Firestore cannot connect to the backend. Could you share some more information about when this error appears? Enabling [logging](https://firebase.google.com/docs/reference/js/firebase.firestore#setloglevel) and sharing the logs, or providing a small sample app with a repro would be helpful for us to debug the issue. Thanks!

**Comment by Sata51** [comment]
I get the same error agains emulator and production. 
On emulator, it appears to failed because auth header is empty (with rules debugging).

```txt
@firebase/firestore: Firestore (8.3.2): Connection WebChannel received: 
```
```json
{
  "targetChange": {
    "targetChangeType": "REMOVE",
    "targetIds": [10],
    "cause": {
      "code": 7,
      "message": "\nNull value error. for 'list' @ L39, Null value error. for 'list' @ L44, Null value error. for 'list' @ L110, Null value error. for 'list' @ L114, Property groups is undefined on object. for 'list' @ L118"
    }
  }
}
```

If I debug my long time working rules, like this:
```
match /users/{userId}/{document=**} {
      allow read: if debug(debug(request.auth.uid) == debug(userId))
}
```
I get this in debug file:
```txt
Apr 08, 2021 1:22:27 PM com.google.net.webchannel.server.common.CorsFilter populateCustomHeaders
WARNING: Invalid $httpHeaders: X-Goog-Api-Client:gl-js/ fire/8.3.2
X-Firebase-GMPID:1:217910399660:web:48a84c58918110a09fc509
Content-Type:text/plain
Authorization:Bearer eyJhbGciOiJSUzI1...
string_value: "FPYDd6wO2NOa0jB40JlCFRXu6K73"
```

By enabling debug on client side, I get no information about this.

**Comment by thebrianchen** [comment]
Closing this since it's a duplicate of #4721. This issue should be fixed in CLI v9.9.0
